roomserver/internal/perform/perform_invite.go

```diff
@@ -54,18 +55,23 @@ func (r *Inviter) PerformInvite(
 		return nil, fmt.Errorf("failed to load RoomInfo: %w", err)
 	}
 
-	log.WithFields(log.Fields{
+	_, domain, _ := gomatrixserverlib.SplitID('@', targetUserID)
+	isTargetLocal := domain == r.Cfg.Matrix.ServerName
+	isOriginLocal := event.Origin() == r.Cfg.Matrix.ServerName
+
+	logger := util.GetLogger(ctx).WithFields(map[string]interface{}{
+		"inviter": event.Sender(),
+		"invitee": *event.StateKey(),
+		"room_id": roomID,
+	})
+	logger.WithFields(log.Fields{
 		"event_id":         event.EventID(),
```

Comment:
Can we move some of these fields into the `logger` definition so that later logging calls have some useful context?